src/components/PreferencesSpot.jsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import createPersistedState from 'use-persisted-state';
+
+import { Add, ErrorOutline, RemoveCircleOutline } from '@mui/icons-material';
+import { Chip, Divider, IconButton, List, ListItem, ListItemIcon, ListItemText, Tooltip, Typography } from '@mui/material';
+
+import { makeStyles } from '@mui/styles';
+import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
+import usePlans from '../hooks/usePlans';
+import useTable from '../hooks/useTable';
+import SpotDialog from './SpotDialog';
+
+const { VITE_TABLE_ID_SPOTS: spotsTableId } = import.meta.env;
+
+const useFavoritesState = createPersistedState('favorites');
+const useStyles = makeStyles({
+  favoriteList: {
+    marginTop: '-8px',
+    paddingBottom: '12px',
+  },
+  favoriteItem: {
+    paddingLeft: '30px',
+    paddingTop: '0',
+    paddingBottom: '0',
+  },
+});
+
+const PreferencesSpot = () => {
+  const [favorites, setFavorites] = useFavoritesState([]);
+  const { place } = useParams();
+  const spots = useTable(Number(spotsTableId));
+  const plans = usePlans();
+  const classes = useStyles();
+
+  const sortedFavorite = plans
+    .map(plan => ({
+      agency: plan.Name,
+      favs: favorites.filter(({ place: favPlace }) => favPlace === plan.Name),
+    }));
+
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+
+  const handleDialogClose = (value, spotPlace) => {
+    setDialogOpen(false);
+    if (!value || favorites.includes(value)) return;
+    setFavorites([
+      ...favorites,
+      {
+        name: value,
+        place: spotPlace,
+      },
+    ]);
+  };
+
+  const removeFavorite = (spotName, spotPlace) => {
+    setFavorites(favorites
+      .filter(fav => (fav.name === spotName && fav.place !== spotPlace) || fav.name !== spotName));
+  };
+
+  const hasFavorite = () =>
+    sortedFavorite.reduce((acc, curr) => acc || curr.favs.length > 0, false);
```

Comment:
```suggestion
  const hasFavorite = () => sortedFavorite.some(e => e.favs.length > 0);
```

un peu plus facile à debug / comprendre